src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -216,18 +288,58 @@ private static bool CommandNamesEqual(ToolCommandName[] left, ToolCommandName[]
             return left.SequenceEqual(right);
         }
 
-        [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
         private class SerializableLocalToolsManifest
         {
-            [DefaultValue(1)]
-            [JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)]
-            public int Version { get; set; }
+            public int? Version { get; set; }
 
             public bool? IsRoot { get; set; }
 
-            [JsonProperty(Required = Required.Always)]
-            // The dictionary's key is the package id
+            /// <summary>
+            /// The dictionary's key is the package id
+            /// this field could be null
+            /// </summary>
             public Dictionary<string, SerializableLocalToolSinglePackage> Tools { get; set; }
+
+            public string ToJson()
+            {
+                var state = new JsonWriterState(options: new JsonWriterOptions { Indented = true });
+                var arrayBufferWriter = new ArrayBufferWriter<byte>();
+                var writer = new Utf8JsonWriter(arrayBufferWriter, state);
+
+                writer.WriteStartObject();
+
+                if (Version.HasValue)
+                {
+                    writer.WriteNumber(propertyName: JsonPropertyVersion, value: Version.Value);
+                }
+
+                if (IsRoot.HasValue)
+                {
+                    writer.WriteBoolean(JsonPropertyIsRoot, IsRoot.Value);
+                }
+
+                writer.WriteStartObject(JsonPropertyTools);
+
+                foreach (var tool in Tools)
+                {
+                    writer.WriteStartObject(tool.Key);
+                    writer.WriteString(JsonPropertyVersion, tool.Value.Version);
+                    writer.WriteStartArray(JsonPropertyCommands);
+                    foreach (var toolCommandName in tool.Value.Commands)
```

Comment:
How large could this list (of commands) be?

Reply:
currently one 1, but it won't be grow too big in the future. Like bundle of tools from one nupkg